### PR TITLE
Spread modifier styles inside _getPopperStyle and update example

### PR DESCRIPTION
**Pull request by brentertz**
#### What does this PR do?

Spreads modifier styles inside _getPopperStyle and updates example.

#### Any background context you want to provide?

We ran into this need when working with a react custom Select component where we want the dropdown list width to match that of the trigger.  

The example includes a simpler scenario where we just change the background color.

#### Screenshot

screen shot 2017-03-09 at 12 32 45 pm

### CODE REVIEW

File: src/Popper.jsx [hunk 1]
```diff
@@ -139,13 +139,15 @@ class Popper extends Component {
...
     } = this.state.data.offsets.popper
+    const styles = this.state.data.styles
 
     return {
       position,
       top:        0,
       left:       0,
       transform:  `translate3d(${left}px, ${top}px, 0px)`,
       willChange: 'transform',
+      ...styles,
```
Comment (brentertz) [inline src/Popper.jsx]: FWIW, this is similar to how PopperJS handles things inside of its own `applyStyle` modifier.

https://github.com/FezVrasta/popper.js/blob/master/src/popper/modifiers/applyStyle.js#L53

Comment (FezVrasta) [inline src/Popper.jsx]: You don't need any of the above definitions if you are going to use the `styles` provided by Popper.js, the code here can be simplified.

**Comment by souporserious** [comment]
I'd like to keep things natural to React where possible. Your use case sounds like it should be solved by just passing a `style` attribute to the `Popper` component. So something like `<Popper style={{ width: variableWidth, backgroundColor: 'red' }}/>`. The style prop gets merged in with the PopperJS styles [here](https://github.com/souporserious/react-popper/blob/master/src/Popper.jsx#L186-L189). Please let me know if that works or not.

**Comment by FezVrasta** [comment]
The advantage of this change is that someone can use custom modifiers of Popper.js with this component. Otherwise one can only add custom modifiers that leverage on `top` and `left` properties.

**Comment by souporserious** [comment]
Sorry, I did some updates. Can you rebase master please?

File: src/Popper.jsx (hunk 1 above)
Comment (souporserious) [inline src/Popper.jsx]: @FezVrasta so are you saying the transform values and such do not need to be added since `styles` is used now?

**Pull request merged**